### ISSUE TITLE
Failures can be created in a broken state

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5077]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1710|
|Type|defect|
|Created|2006-05-09 13:56:52Z|

Creating a Failure with no explicit exception when sys.exc_info() returns None will produce an object that is unusable in various ways.  This should not be allowed.

<details><summary>Searchable metadata</summary>

```
trac-id__1710 1710
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1147183012000000 1147183012000000
changetime__1147441823000000 1147441823000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Ready for review in broken-failures-1710

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

Hooray unit tests

+1

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16775]) Merge broken-failure-1710

Author: exarkun
Reviewer: radix
Fixes [#5077](https://github.com/twisted/twisted/issues/5077)

Cause Failure to synchronously raise an exception if constructed with no exception state.
Also remove the special handling of this case in the debug mode of Failure.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
